packages/pigeon/lib/swift_generator.dart

```diff
@@ -739,9 +738,9 @@ String _flattenTypeArguments(List<TypeDeclaration> args) {
 String _swiftTypeForBuiltinGenericDartType(TypeDeclaration type) {
   if (type.typeArguments.isEmpty) {
     if (type.baseName == 'List') {
-      return '[Any]';
+      return '[Any?]';
     } else if (type.baseName == 'Map') {
-      return '[AnyHashable: Any]';
+      return '[AnyHashable: Any?]';
     } else {
       return 'Any';
```

Comment:
This case (and the corresponding case in Line 753) is not used, since the caller only call this function with `List/Map` types: 

```
  if (type.baseName == 'List' || type.baseName == 'Map') {
    return _swiftTypeForBuiltinGenericDartType(type);
  }
```

I would either write an assert, or split it into 2 separate helpers (for list and map). But for now I'll just leave it here to keep this PR focused.